sslCheck/sslCheck_test.go

```diff
@@ -53,112 +51,121 @@ func TestCheckHost(t *testing.T) {
-		shouldReturnCertErr: true,
-		expectedCertErr: certError{
-			ExpiresInFiveDaysOrLess:    false,
-			ExpiresInFifteenDaysOrLess: false,
-			ExpiresInThirtyDaysOrLess:  true,
-			ExpiresInSixtyDaysOrLess:   true,
+		{
+			description:         "When cert is valid for 120",
+			validForDays:        120,
+			listenPort:          ":9447",
+			host:                "127.0.0.1:9447",
+			hostHasError:        false,
+			shouldReturnCertErr: false,
 		},
-	},
-	{
-		description:         "When cert is valid for 60",
-		validForDays:        60,
-		listenPort:          ":9446",
-		host:                "127.0.0.1:9446",
-		hostHasError:        false,
-		shouldReturnCertErr: true,
-		expectedCertErr: certError{
-			ExpiresInFiveDaysOrLess:    false,
-			ExpiresInFifteenDaysOrLess: false,
-			ExpiresInThirtyDaysOrLess:  false,
-			ExpiresInSixtyDaysOrLess:   true,
+		{
+			description:         "When cert is not valid",
+			validForDays:        0,
+			listenPort:          ":9448",
+			host:                "127.0.0.1:9448",
+			hostHasError:        true,
+			shouldReturnCertErr: false,
 		},
-	},
-	{
-		description:         "When cert is valid for 120",
-		validForDays:        120,
-		listenPort:          ":9447",
-		host:                "127.0.0.1:9447",
-		hostHasError:        false,
-		shouldReturnCertErr: false,
-	},
-	{
-		description:         "When cert is not valid",
-		validForDays:        0,
-		listenPort:          ":9448",
-		host:                "127.0.0.1:9448",
-		hostHasError:        true,
-		shouldReturnCertErr: false,
-	},*/
 	}
 
 	for _, test := range tests {
-		g.Describe("ProcessHosts()", func() {
-			serverCrt, CAPem := GenerateCertBundle(test.validForDays)
-			g.Before(func() {
-				go func() {
-					config := &tls.Config{Certificates: []tls.Certificate{serverCrt}}
-					ln, err := tls.Listen("tcp", test.listenPort, config)
-					if err != nil {
-						fmt.Printf("Error listening on port  %v err: %v", test.listenPort, err)
-					}
-					conn, err := ln.Accept()
-					if err != nil {
-						fmt.Printf("Error Accepting Conn %v", err)
-					}
-					conn.Write([]byte("test\n"))
-					conn.Close()
-					ln.Close()
-				}()
-			})
-			g.It(test.description, func() {
-				result := checkHost(test.host, CAPem)
-				fmt.Printf("result %s", result)
-				g.Assert(result.Err != nil).Equal(test.hostHasError)
-				if test.shouldReturnCertErr {
-					g.Assert(result.CertErrors[0].ExpiresInFifteenDaysOrLess).Equal(test.expectedCertErr.ExpiresInFifteenDaysOrLess)
-					g.Assert(result.CertErrors[0].ExpiresInFiveDaysOrLess).Equal(test.expectedCertErr.ExpiresInFiveDaysOrLess)
-					g.Assert(result.CertErrors[0].ExpiresInSixtyDaysOrLess).Equal(test.expectedCertErr.ExpiresInSixtyDaysOrLess)
-					g.Assert(result.CertErrors[0].ExpiresInThirtyDaysOrLess).Equal(test.expectedCertErr.ExpiresInThirtyDaysOrLess)
-				}
-			})
-		})
+		serverCrt, CAPem := GenerateCertBundle(test.validForDays)
+
+		nearlyReady := make(chan bool,1 )
+		go func(ready chan<- bool) {
```

Comment:
This is really cool, and makes sense now that I see it written out, but I wonder why it worked so consistently locally but not on jenkins.

Reply:
My assumption about why it wasn't working consistently is that it took inconsistent amount of time for connection to start up and listen. Basically on Jenkins it would sometimes be trying to startup the go routine and begin listening when the checkHost call was made and would result in failure. What I did here is force everything to wait until nearly all the listening steps are done then let the test continue.